src/util/innerStore.ts

```diff
@@ -51,4 +51,29 @@ export interface ECData {
         option: ComponentItemTooltipOption<unknown>;
     };
 }
+
+const setItemDataAndSeriesIndex = function (this: Element, child: Element): void {
+    const childECData = getECData(child);
+    const thisECData = getECData(this);
```

Comment:
I think `this` context is wrong here.  It's not the root element in `setCommenECData`

Use anonymous function in `setCommonECData` is fine. JS engine will handle it well and won't have a performance issue. Use a separate function with a specified `this` context will be easy to have the above issue.